File: packages/main/update.ts
import { autoUpdater } from "electron"

export function update(): void {
    autoUpdater.setFeedURL({ url: "http://localhost:3000/"})
    autoUpdater.on("error", (err) => {
        console.log("autoupdater failed:", err)
    })
    autoUpdater.on("checking-for-update", () => {
        console.log("Checking for update:")
    })
    setInterval(() => {
        autoUpdater.checkForUpdates()
    }, 60000)
}

File: packages/backend/update.go
package main

import (
	"log"
	"net/http"

	"github.com/minio/selfupdate"
)

func doUpdate(url string) error {
	log.Println("Updating from:", url)
	resp, err := http.Get(url)
	if err != nil {
		log.Fatal(err)
		return err
	}
	defer resp.Body.Close()

	err = selfupdate.Apply(resp.Body, selfupdate.Options{})
	if err != nil {
		log.Println("Failed update")
	}
	log.Println("update successful")
	return err
}
